src/core/store-data.ts

```diff
@@ -1,56 +1,223 @@
+    const author =
+      authorName && authorEmail
+        ? {
+            name: authorName,
+            email: authorEmail,
+          }
+        : undefined;
+
+    const committerName = getOptionalEnvVar("COMMITTER_NAME");
+    const committerEmail = getOptionalEnvVar("COMMITTER_EMAIL");
+    const committer =
+      committerName && committerEmail
+        ? {
+            name: committerName,
+            email: committerEmail,
+          }
+        : undefined;
+
+    /*
+      [Step 3/4]
+      
+      Post a new git commit object to GitHub.
+      The author and committer may be undefined.
+      The parent commit is set to the current branch HEAD.
+      The tree is our newly created tree with all the file contents.
+    */
+    const {
+      data: { sha: newCommitSha },
+    } = await octokitForDataRepo.request(
+      "POST /repos/{owner}/{repo}/git/commits",
+      {
+        owner,
+        repo,
+        message: `auto(data): add metrics from ${commitMessageTrigger} for ${triggerOwner}/${triggerRepo}`,
+        author,
+        committer,
+        parents: [headShas.commitSha],
+        tree: newTreeSha,
+      }
     );
 
-    const identifier = `${data.trigger_event_signature} => ${data.metric_signature}`;
-    const path = `${process.env.REPO_PATH}/${data.owner}/${data.repo}/${data.metric_signature}/${data.created_at}.json`;
+    /*
+      [Step 4/4]
+      
+      Set the target branch to our newly created ref.
+      `force` is set to false, so we do not overwrite another commit that may
+      have been committed in the meantime (equivalent to not force-pushing).
+
+      If there is a newer commit by now, we try to fetch the new commit SHAs and
+      do evertyhing again. If there isn't, or if we're out of retries, we just
+      throw the error.
+      Note that there is no way to remove the now unused objects on GitHub, but
+      they will probably expire at some point.
+    */
     try {
       await octokitForDataRepo.request(
-        "PUT /repos/{owner}/{repo}/contents/{path}",
+        "PATCH /repos/{owner}/{repo}/git/refs/{ref}",
         {
-          owner: process.env.REPO_OWNER as string,
-          repo: process.env.REPO_NAME as string,
-          path,
-          message: `auto(data): add ${identifier} for ${data.owner}/${data.repo}`,
-          committer: {
-            name: process.env.COMMITTER_NAME as string,
-            email: process.env.COMMITTER_EMAIL as string,
-          },
-          author: {
-            name: process.env.AUTHOR_NAME as string,
-            email: process.env.AUTHOR_EMAIL as string,
-          },
-          content: Base64.encode(JSON.stringify(data)),
+          owner,
+          repo,
+          ref: `heads/${targetBranch}`,
+          sha: newCommitSha,
+          force: false,
         }
       );
+    } catch (e: unknown) {
+      if (tryNumber === retriesAllowed) {
+        // out of retries, just throw the error
```

Comment:
maybe we can also add a log here to let us know we reached the maximum retries. I would even add the value of `retriesAllowed` to the log message.